TrackletAlgorithm/MemoryTemplateTPROJ.h

```diff
@@ -68,13 +85,14 @@ class MemoryTemplateTPROJ
     return mask_[bx];
   }
   
-  const DataType (&get_mem() const)[1<<NBIT_BX][(1<<NBIT_ADDR)*NPAGE] {return dataarray_;}
+  const DataType (&get_mem() const)[DEPTH_BX][(DEPTH_ADDR)*NPAGE] {return dataarray_;}
 
   DataType read_mem(BunchXingT ibx, ap_uint<NBIT_ADDR> index, unsigned int page = 0) const
   {
-	// TODO: check if valid
-	if(!NBIT_BX) ibx = 0;
-	return dataarray_[ibx][(1<<NBIT_ADDR)*page+index];
+    //assert(page < NPAGE);    
+  	// TODO: check if valid  
+	  if(!NBIT_BX) ibx = 0;
+ 	  return dataarray_[ibx][DEPTH_ADDR*page+index];
```

Comment:
@tomalin Same comment regarding indentation.

```suggestion
    //assert(page < NPAGE);
    // TODO: check if valid
    if(!NBIT_BX) ibx = 0;
    return dataarray_[ibx][DEPTH_ADDR*page+index];
```